streaming/src/main/scala/org/apache/spark/streaming/dstream/ReceiverInputDStream.scala

```diff
@@ -116,7 +116,10 @@ abstract class ReceiverInputDStream[T: ClassTag](@transient ssc_ : StreamingCont
             logWarning("Some blocks have Write Ahead Log information; this is unexpected")
           }
         }
-        new BlockRDD[T](ssc.sc, blockIds)
+        val validBlockIds = blockIds.filter { id =>
+          ssc.sparkContext.env.blockManager.master.contains(id)
```

Comment:
Is it worth to add some warning log here? I think the user may forget to enable receiver log.